charts/datadog/templates/agent-clusterchecks-rbac.yaml

```
@@ -14,6 +14,23 @@ subjects:
     name: {{ template "datadog.fullname" . }}-cluster-checks
     namespace: {{ .Release.Namespace }}
 ---
+{{- if and .Values.datadog.kubeStateMetricsCore.enabled .Values.datadog.kubeStateMetricsCore.useClusterCheckWorkers -}}
+apiVersion: {{ template "rbac.apiVersion" . }}
+kind: ClusterRoleBinding
+metadata:
+  labels:
+{{ include "datadog.labels" . | indent 4 }}
+  name: {{ template "datadog.fullname" . }}-cluster-checks-ksm-core
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: {{ template "datadog.fullname" . }}-ksm-core
+subjects:
+  - kind: ServiceAccount
+    name: {{ template "datadog.fullname" . }}-cluster-checks
+    namespace: {{ .Release.Namespace }}
+---
+{{- end }}
```

We could factorise the creation of the `ClusterRoleBinding` for KSM to have only one definition but with a condition for the `ServiceAccount`.

```suggestion
{{- if and .Values.datadog.kubeStateMetricsCore.enabled  -}}
---
apiVersion: {{ template "rbac.apiVersion" . }}
kind: ClusterRoleBinding
metadata:
  labels:
{{ include "datadog.labels" . | indent 4 }}
  name: {{ template "datadog.fullname" . }}-ksm-core
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: {{ template "datadog.fullname" . }}-ksm-core
subjects:
  - kind: ServiceAccount
    {{- if .Values.datadog.kubeStateMetricsCore.useClusterCheckWorkers }}
    name: {{ template "datadog.fullname" . }}-cluster-checks
    {{- else }}
    name: {{ template "datadog.fullname" . }}-cluster-agent
    {{- endif }}
    namespace: {{ .Release.Namespace }}
{{- end }}
---
```

But in this case, this “generic” `ClusterRoleBinding` definition should move to `charts/datadog/templates/kube-state-metrics-core-cr.yaml`.